### PR TITLE
feat(plugin): add publish button to data catalog

### PR DESCRIPTION
**Pull request by nourbalaha**
Done:
- Add publish button to data catalog's dataset tab
- Refactor styled components

Not done yet:
- Add functionality to publish button in data catalog's dataset tab

### CODE REVIEW

File: plugin/web/extensions/sidebar/modals/datacatalog/components/content/DatasetDetails.tsx [hunk 1]
```diff
@@ -19,61 +19,124 @@ const DatasetDetails: React.FC<Props> = ({
...
+            {!addDisabled && <Icon icon="plusCircle" />}
+            {addDisabled ? "シーンに追加済み" : "シーンに追加"}
+          </AddButton>
+          <ShareButton isShareable={isShareable}>
+            <Icon icon="share" />
+            シェア
+          </ShareButton>
+        </ButtonWrapper>
+      </TopWrapper>
+      <Wrapper>{ContentSection && <ContentSection />}</Wrapper>
```
Comment (KaWaite) [inline plugin/web/extensions/sidebar/modals/datacatalog/components/content/DatasetDetails.tsx:55]: Unless you need the Wrapper for the structure of the details, can you rewrite like:
```
{ContentSection && (
   <Wrapper>
     <ContentSection />
  </Wrapper>
  )
}
```

File: plugin/web/extensions/sidebar/modals/datacatalog/components/content/DatasetDetails.tsx [hunk 2]
```diff
@@ -19,61 +19,124 @@ const DatasetDetails: React.FC<Props> = ({
...
-      </ButtonWrapper>
-      {ContentSection && <ContentSection />}
-    </Wrapper>
+    <>
+      <TopWrapper>
+        <HeaderWrapper>
+          <Title>{dataset.type === "item" && (dataset.cityName ?? dataset.name)}</Title>
+          <PublishButton onClick={handlePublish} published={published} isShareable={isShareable}>
+            <HoverText published={published}>Publish</HoverText>
+            <Text published={published}>{published ? "Published" : "Unpublished"}</Text>
```
Comment (KaWaite) [inline plugin/web/extensions/sidebar/modals/datacatalog/components/content/DatasetDetails.tsx:41]: Published = 公開済み
Unpublished = 未公開

**Review by KaWaite — APPROVED** [review]

File: plugin/web/extensions/sidebar/modals/datacatalog/components/content/DatasetDetails.tsx (hunk 2 above)
Comment (nourbalaha) [reply plugin/web/extensions/sidebar/modals/datacatalog/components/content/DatasetDetails.tsx:41]: Done!

File: plugin/web/extensions/sidebar/modals/datacatalog/components/content/DatasetDetails.tsx (hunk 1 above)
Comment (nourbalaha) [reply plugin/web/extensions/sidebar/modals/datacatalog/components/content/DatasetDetails.tsx:55]: Done!

**Pull request merged**